# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/chrome-webidl/BrowsingContext.webidl
# Commit: de69e3c1635a
# Full Hash: de69e3c1635a311b5b22716b65b93fc1899a37a4
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-08-01 09:48:10
# Regressor Bug: 1613431
# File Overlap Count: 1
# Description:
#   Bug 1613431 - Part 3: Handle synced setters return value. r=nika
#   
#   Depends on D83646
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83647
# ==============================================================================

diff -r b326d33f98cb -r de69e3c1635a dom/chrome-webidl/BrowsingContext.webidl
--- a/dom/chrome-webidl/BrowsingContext.webidl	Fri Jul 31 13:37:00 2020 +0000
+++ b/dom/chrome-webidl/BrowsingContext.webidl	Fri Jul 31 13:37:13 2020 +0000
@@ -23,7 +23,7 @@
 
   readonly attribute boolean useRemoteSubframes;
 
-  [BinaryName="useTrackingProtectionWebIDL"]
+  [BinaryName="useTrackingProtectionWebIDL", SetterThrows]
   attribute boolean useTrackingProtection;
 
   [NewObject, Throws]
@@ -68,9 +68,9 @@
 
   readonly attribute WindowContext? topWindowContext;
 
-  attribute [TreatNullAs=EmptyString] DOMString customPlatform;
+  [SetterThrows] attribute [TreatNullAs=EmptyString] DOMString customPlatform;
 
-  attribute [TreatNullAs=EmptyString] DOMString customUserAgent;
+  [SetterThrows] attribute [TreatNullAs=EmptyString] DOMString customUserAgent;
 
   readonly attribute DOMString embedderElementType;
 
@@ -103,11 +103,11 @@
 
   // Extension to give chrome JS the ability to set the window screen
   // orientation while in RDM.
-  void setRDMPaneOrientation(OrientationType type, float rotationAngle);
+  [Throws] void setRDMPaneOrientation(OrientationType type, float rotationAngle);
 
   // Extension to give chrome JS the ability to set a maxTouchPoints override
   // while in RDM.
-  void setRDMPaneMaxTouchPoints(octet maxTouchPoints);
+  [Throws] void setRDMPaneMaxTouchPoints(octet maxTouchPoints);
 
   // The watchedByDevTools flag indicates whether or not DevTools are currently
   // debugging this browsing context.
@@ -143,7 +143,7 @@
   readonly attribute WindowGlobalParent? embedderWindowGlobal;
 
   void notifyStartDelayedAutoplayMedia();
-  void notifyMediaMutedChanged(boolean muted);
+  [Throws] void notifyMediaMutedChanged(boolean muted);
 
   readonly attribute nsISecureBrowserUI? secureBrowserUI;
 